src/ds/champ_map_serializers.h

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf_assert.h"
+#include "serialized.h"
+
+#include <msgpack/msgpack.hpp>
+#include <nlohmann/json.hpp>
+#include <small_vector/SmallVector.h>
+
+namespace champ
+{
+  using Version = int64_t;
+
+  template <typename V>
+  struct VersionV
+  {
+    Version version;
+    V value;
+
+    VersionV() = default;
+    VersionV(Version ver, V val) : version(ver), value(val) {}
+  };
+
+  namespace serialisers
+  {
+    using SerialisedEntry = llvm_vecsmall::SmallVector<uint8_t, 8>;
+  }
+
+  namespace untyped
+  {
+    using SerialisedEntry = champ::serialisers::SerialisedEntry;
+    using VersionV = champ::VersionV<SerialisedEntry>;
+  }
+
+  template <class T>
+  inline size_t get_size(const T& data)
+  {
+    return sizeof(uint64_t) + sizeof(T);
+  }
+
+  template <>
+  inline size_t get_size<champ::untyped::SerialisedEntry>(
+    const champ::untyped::SerialisedEntry& data)
+  {
+    return sizeof(uint64_t) + data.size();
+  }
+
+  template <>
+  inline size_t get_size<champ::untyped::VersionV>(
+    const champ::untyped::VersionV& data)
+  {
+    return sizeof(uint64_t) + sizeof(data.version) + data.value.size();
+  }
+
+  template <class T>
+  inline size_t serialize(const T& t, uint8_t*& data, size_t& size)
+  {
+    uint64_t data_size = sizeof(T);
+    serialized::write(
+      data, size, reinterpret_cast<const uint8_t*>(&data_size), sizeof(T));
```

Comment:
Not that it matters here but the the last parameter should be `sizeof(uint64_t)`?